sample/java/app/controllers/Application.java

```diff
@@ -24,39 +28,30 @@ public Result index() {
     return ok(index.render("Your new application is ready."));
   }
 
-  public F.Promise<Result> getCountry(final String addr) {
-    return this.geolocationService.getGeolocation(addr)
-        .map(new F.Function<Geolocation, Result>() {
-          public Result apply(Geolocation geolocation) {
-            String country = geolocation.getCountryName();
-            String message = country != null ? String.format("This ip comes from %s", country) :
-                "Sorry, we couldn't connect to the webservice";
-            return ok(index.render(message));
-          }
-        });
+  public CompletionStage<Result> getCountry(final String addr) {
+    return geolocationService.getGeolocation(addr)
+                             .thenApplyAsync(geolocation -> ok(index.render(formHeader(geolocation, addr, Geolocation::getCountryName))));
+  }
+
+  public CompletionStage<Result> getCountryCode(final String addr) {
+    return geolocationService.getGeolocation(addr)
+        .thenApplyAsync(geolocation -> ok(index.render(formHeader(geolocation, addr))));
+  }
+
+  public CompletionStage<Result> getGeolocation(final String addr) {
+    return geolocationService.getGeolocation(addr)
+        .thenApplyAsync(geolocation -> ok(geoData.render(geolocation, formHeader(geolocation, addr))));
   }
 
-  public F.Promise<Result> getCountryCode(final String addr) {
-    return this.geolocationService.getGeolocation(addr)
-        .map(new F.Function<Geolocation, Result>() {
-          public Result apply(Geolocation geolocation) {
-            String countryCode = geolocation.getCountryCode();
-            String message = countryCode != null ? String.format("This ip comes from %s", countryCode) :
-                "Sorry, we couldn't connect to the webservice";
-            return ok(index.render(message));
-          }
-        });
+  private String formHeader(Geolocation geolocation, String addr)
+  {
+    return formHeader(geolocation, addr, Geolocation::getCountryCode);
   }
 
-  public F.Promise<Result> getGeolocation(final String addr) {
-    return this.geolocationService.getGeolocation(addr)
-        .map(new F.Function<Geolocation, Result>() {
-          public Result apply(Geolocation geolocation) {
-            String countryCode = geolocation.getCountryCode();
-            String message = countryCode != null ? String.format("This ip comes from %s", countryCode) :
-                "Sorry, we couldn't connect to the webservice";
-            return ok(geoData.render(geolocation));
-          }
-        });
+  private String formHeader(Geolocation geolocation, String addr, Function<Geolocation, String> propertyAccessor)
+  {
```

Comment:
I'm nitpicking here: please follow the current code style:

```java
private String formHeader(Geolocation geolocation, String addr, Function<Geolocation, String> propertyAccessor) {
...
```
Instead of:

```java
private String formHeader(Geolocation geolocation, String addr, Function<Geolocation, String> propertyAccessor) 
{ // new line
```